### PR TITLE
Add initialize/stop idle-timers when mode on/off

### PR DESCRIPTION
**Pull request by marcwebbie**

### CODE REVIEW

File: super-save.el [hunk 1]
```diff
@@ -82,15 +82,29 @@ See `super-save-auto-save-when-idle'."
...
+(defun super-save-initialize-idle-timer ()
+  "Initialize super-save idle timer if `super-save-auto-save-when-idle' is true."
+  (setq super-save-idle-timer
+        (when super-save-auto-save-when-idle
+          (run-with-idle-timer super-save-idle-duration t #'super-save-command))))
+
+(defun super-save-stop-idle-timer ()
+  "Stop super-save idle timer if `super-save-idle-timer' is set."
+  (when super-save-idle-timer
+    (setq super-save-idle-timer
```
Comment (bbatsov) [inline super-save.el]: This will return `nil`, so the `setq` is not necessary.

Comment (bbatsov) [inline super-save.el]: This = the `cancel-timer` function.

**Comment by bbatsov** [comment]
That's definitely the right approach, but one small thing remains - the timer var should be simply set to `nil` where it's declared (as now it's initialized in the new function you've created).

**Comment by marcwebbie** [comment]
@bbatsov thanks for the suggestions.

**Comment by bbatsov** [comment]
Looks good! Can you just squash the 3 commits into one?

**Comment by marcwebbie** [comment]
@bbatsov commits squashed. :)

**Comment by bbatsov** [comment]
:+1:

**Pull request merged**